cmd/snap/cmd_snap_op.go

```diff
@@ -826,17 +828,18 @@ func (x *cmdRefresh) Execute([]string) error {
 	}
 
 	names := installedSnapNames(x.Positional.Snaps)
+	opts := &client.SnapOptions{
+		Amend:            x.Amend,
+		Channel:          x.Channel,
+		IgnoreValidation: x.IgnoreValidation,
+		IgnoreRunning:    x.IgnoreRunning,
+		Revision:         x.Revision,
+		CohortKey:        x.Cohort,
+		LeaveCohort:      x.LeaveCohort,
+		Transactional:    x.Transactional,
+	}
+	x.setModes(opts)
 	if len(names) == 1 {
-		opts := &client.SnapOptions{
```

Comment:
I wonder if it wouldn't be clearer if we had:
```
if len(names) == 1 {
	opts := &client.SnapOptions{
		Amend:            x.Amend,
		Channel:          x.Channel,
		IgnoreValidation: x.IgnoreValidation,
		IgnoreRunning:    x.IgnoreRunning,
		Revision:         x.Revision,
		CohortKey:        x.Cohort,
		LeaveCohort:      x.LeaveCohort,
		Transactional:    x.Transactional,
	}
	x.setModes(opts)
	return x.refreshOne(names[0], opts)
}
// only transactional flag has meaning when refreshing
// with many snaps
opts := &client.SnapOptions{
	Transactional:    x.Transactional,
}
...
```

Reply:
I think I agree with this suggestion